packages/features/insights/context/FiltersProvider.tsx

```diff
@@ -117,18 +117,21 @@ export function FiltersProvider({ children }: { children: React.ReactNode }) {
         setSelectedTimeView: (selectedTimeView) => setSelectedTimeView(selectedTimeView),
         setSelectedMemberUserId: (selectedMemberUserId) => {
           setSelectedMemberUserId(selectedMemberUserId);
+          const { userId, eventTypeId, ...rest } = router.query;
```

Comment:
Fixes some scenarios that should remove queries from url.

Reply:
Would useTypedQuery help here? Essentially does this but with some helper functions

Reply:
useTypedQuery can't handle multiple changes from one call. I've tried and it was very flaky.